### PR TITLE
Update subscriptions-core version 2.4.0

### PR DESCRIPTION
**Pull request by mattallan**
#### Changes proposed in this Pull Request

This PR bumps the `subscriptions-core` version in WooCommerce Payments to 2.4.0 and adds any relevant changelog entries.

This new version of Subscriptions Core includes changes to our `wcs_create_subscription()` in preparation for supporting High-Performance Order Storage (HPOS).

More details can be found in the merged PRs over on the `subscriptions-core` repository:
- https://github.com/Automattic/woocommerce-subscriptions-core/pull/226
- https://github.com/Automattic/woocommerce-subscriptions-core/pull/228

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : Additional QA Testing Not Applicable
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: changelog/subscriptions-core-2.4.0-2 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: update
+
+ImImprove wcs_copy_order_address() to use modern APIs for setting address fields.
```
Comment (Jinksi) [inline changelog/subscriptions-core-2.4.0-2]:
```suggestion
Improve wcs_copy_order_address() to use modern APIs for setting address fields.
```

Nitpick: typo

Comment (mattallan) [reply changelog/subscriptions-core-2.4.0-2]: How did I fail at copy-pasting? haha

Comment (mattallan) [reply changelog/subscriptions-core-2.4.0-2]: Thanks for spotting this!

**Review by Jinksi — APPROVED** [review]
LGTM 👍

- [x] Post-checkout hook installed subs-core 2.4.0.
- [x] Vendor subs-core directory updated with correct subs-core version.
- [x] Complete purchase of subscription product as shopper.

**Pull request merged**